manage-data/ingest/transform-enrich/readable-maintainable-ingest-pipelines.md

```diff
@@ -0,0 +1,677 @@
+
+---
+mapped_pages:
+  - https://www.elastic.co/docs/manage-data/ingest/transform-enrich/common-mistakes.html
+applies_to:
+  stack: ga
+  serverless: ga
+---
+
+# Create readable and maintainable ingest pipelines
+
+There are many ways to achieve similar results when creating ingest pipelines, which can make maintenance and readability difficult. This guide outlines patterns you can follow to make the maintenance and readability of ingest pipelines easier without sacrificing functionality.
+
+:::{note}
+This guide does not provide guidance on optimizing for ingest pipeline performance.
+:::
+
+## Access fields
+
+When creating ingest pipelines, there are are few options for accessing fields in conditional statements and scripts. All formats can be used to reference fields, so choose the one that makes your pipeline easier to read and maintain.
+
+| Notation                       | Example                                               | Notes                                                                           |
+| ------------------------------ | ----------------------------------------------------- | ------------------------------------------------------------------------------- |
+| Dot notation                   | `ctx.event.action`                                    | Supported in conditionals and painless scripts.                                 |
+| Square bracket notation        | `ctx['event']['action']`                              | Supported in conditionals and painless scripts.                                 |
+| Mixed dot and bracket notation | `ctx.event['action']`                                 | Supported in conditionals and painless scripts.                                 |
+| Field API                      | `field('event.action', '')` or `$('event.action','')` | Supported in conditionals and painless scripts. Only available in versions 9.2+ |
```

Comment:
Use an `applies_to` badge. This will be rendered on the page as `Planned` until the 9.2.0 release date. 

```suggestion
| Field API  {applies_to}`stack: ga 9.2.0`                    | `field('event.action', '')` or `$('event.action','')` | Supported in conditionals and painless scripts. |
```